curator-recipes/src/main/java/org/apache/curator/framework/recipes/leader/LeaderLatch.java

```diff
@@ -607,6 +607,17 @@ private void getChildren() throws Exception {
             public void processResult(CuratorFramework client, CuratorEvent event) throws Exception {
                 if (event.getResultCode() == KeeperException.Code.OK.intValue()) {
                     checkLeadership(event.getChildren());
+                } else if (event.getResultCode() == KeeperException.Code.NONODE.intValue()) {
+                    // latchPath has gone - reset
+                    //
+                    // This is possible when RECONNECTED during:
+                    // (1) Scale the zk cluster to 0 nodes.
+                    // (2) Scale it back.
+                    //
+                    // See also https://issues.apache.org/jira/browse/CURATOR-724
+                    reset();
```

Comment:
> BTW, before https://github.com/apache/curator/pull/430, the leaderLatch calls reset method after reconnection, and reset will whatever create the missing parent path nodes. So I didn't find this issue in the old versions.   -- [CURATOR-724](https://issues.apache.org/jira/browse/CURATOR-724?focusedCommentId=17901058&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17901058)

Just for sure, so, in case of `RECONNECTED`, we have no idea what happened to the cluster and `getChildren` is a conservative option comparing to `reset`.

Reply:
Yeah. Updated add a test.

Reply:
Also ensure that the test failed without this fix.